Fix xglock.c errors

Newer versions of the C compiler will fail on both:
  This trick that used to be used to indicate a return type was being
  intentionally ignored:
    (void) g_free(udata);
  A type mismatch in the prototypes, because the __PROTO macro ends up
  skipping the prototypes and just giving empty function arguments for
  all arguments.
  Note that this problem shows up in Gentoo Bug #949219.  One potential
  fix is to compile with -std=gnu17, but that does not fix the first
  problem, so try a patch to fix things properly.
Fix the trick by removing the (void) cast.  Fix the prototype conflicts
by commenting out the prototypes that are either duplicates or conflict
with existing prototypes.

diff --git a/xglock/xglock.c b/xglock/xglock.c
index 1234567..1234567 100644
--- a/xglock/xglock.c
+++ b/xglock/xglock.c
@@ -114,11 +114,16 @@ static GtkTooltips *tooltips=(GtkTooltips *)NULL;
 
 #define STD_CB_ARGS	(GtkWidget *, gpointer)
 
+/* These already have prototypes.  Comment out to avoid conflicts. */
+/*
 static void *__PROTO( secured_malloc, (size_t) );
 static void *__PROTO( secured_calloc, (int, size_t) );
 static void *__PROTO( secured_realloc, (void *, size_t) );
 static gchar *__PROTO( check_quotes, (gchar *str) );
+*/
 
+/* These already have prototypes.  Comment out to avoid conflicts. */
+/*
 static void __PROTO( mode_list_item_select_cb, STD_CB_ARGS );
 static void __PROTO( mode_list_item_unselect_cb, STD_CB_ARGS );
 static void __PROTO( compose_xlock_command, (char *) );
@@ -127,33 +132,52 @@ static void __PROTO( launch_xlock, STD_CB_ARGS );
 static void __PROTO( exit_xglock, STD_CB_ARGS );
 static void __PROTO( bool_option_clicked_cb, STD_CB_ARGS );
 static void __PROTO( destroy_window, STD_CB_ARGS );
+*/
 
+/* These already have prototypes.  Comment out to avoid conflicts. */
+/*
 static void __PROTO( color_selection_cancel_cb, STD_CB_ARGS );
 static void __PROTO( color_selection_ok_cb, STD_CB_ARGS );
 static void __PROTO( color_selection_changed_cb, STD_CB_ARGS );
 static void __PROTO( create_color_selection_dialog, STD_CB_ARGS );
+*/
 
 /*static void __PROTO( font_select_item_select_cb, STD_CB_ARGS );*/
+/* These already have prototypes.  Comment out to avoid conflicts. */
+/*
 static void __PROTO( font_select_cancel_cb, STD_CB_ARGS );
 static void __PROTO( font_select_ok_cb, STD_CB_ARGS );
 static void __PROTO( create_font_selection_dialog, STD_CB_ARGS );
+*/
 
+/* These already have prototypes.  Comment out to avoid conflicts. */
+/*
 static void __PROTO( file_selection_cancel_cb, STD_CB_ARGS );
 static void __PROTO( file_selection_ok_cb, STD_CB_ARGS );
 static void __PROTO( create_file_selection_dialog, STD_CB_ARGS );
+*/
 
+/* These already have prototypes.  Comment out to avoid conflicts. */
+/*
 static void __PROTO( contextual_help_dialog, STD_CB_ARGS );
 static void __PROTO( window_help_dialog, STD_CB_ARGS );
 static void __PROTO( global_help_dialog, STD_CB_ARGS );
 static void __PROTO( about_dialog, STD_CB_ARGS );
 static void __PROTO( about_dialog_close_callback, STD_CB_ARGS );
+*/
 
+/* These already have prototypes.  Comment out to avoid conflicts. */
+/*
 static GtkWidget * __PROTO( create_file_menu, (GtkWidget *) );
 static GtkWidget * __PROTO( create_help_menu, (GtkWidget *) );
+*/
 
+/* These already have prototypes.  Comment out to avoid conflicts. */
+/*
 static void __PROTO( create_fntColorOptions_entries, (GtkWidget *) );
 static void __PROTO( create_genOptions_entries, (GtkWidget *) );
 static void __PROTO( create_boolOptions_buttons, (GtkWidget *) );
+*/
 
 #if GTK_CHECK_VERSION(2,0,0)
 /* GTK+ 2.x differences */
@@ -714,7 +738,7 @@ color_selection_cancel_cb(GtkWidget *w,
     struct_option_fntcol_callback *cbs_ptr =(struct_option_fntcol_callback *)udata;
     gtk_widget_hide(cbs_ptr->fntcol_dialog);
     gtk_widget_destroy(cbs_ptr->fntcol_dialog);
-    (void) g_free(udata);
+    g_free(udata);
 }
 
 /*
@@ -777,7 +801,7 @@ color_selection_ok_cb(GtkWidget *w,
     gtk_entry_set_text( GTK_ENTRY(cbs_ptr->entry), colorstr );
     gtk_widget_hide( cbs_ptr->fntcol_dialog );
     gtk_widget_destroy(cbs_ptr->fntcol_dialog);
-    (void) g_free(udata);
+    g_free(udata);
 }
 
 /*
@@ -910,7 +934,7 @@ font_select_cancel_cb(GtkWidget *w,
     struct_option_fntcol_callback *cbs_ptr = (struct_option_fntcol_callback *)udata;
     gtk_widget_hide(cbs_ptr->fntcol_dialog);
     gtk_widget_destroy(cbs_ptr->fntcol_dialog);
-    (void) g_free(udata);
+    g_free(udata);
 }
 
 /*
@@ -927,7 +951,7 @@ font_select_ok_cb(GtkWidget *w,
     gtk_entry_set_text(GTK_ENTRY(cbs_ptr->entry),
                        gtk_font_selection_dialog_get_font_name(GTK_FONT_SELECTION_DIALOG(cbs_ptr->fntcol_dialog)));
     gtk_widget_destroy(cbs_ptr->fntcol_dialog);
-    (void) g_free(udata);
+    g_free(udata);
 }
 
 /*
@@ -979,7 +1003,7 @@ file_selection_cancel_cb(GtkWidget *w,
     struct_option_fntcol_callback *cbs_ptr= (struct_option_fntcol_callback *)udata;
     gtk_widget_hide( cbs_ptr->fntcol_dialog );
     gtk_widget_destroy( cbs_ptr->fntcol_dialog );
-    (void) g_free(udata);
+    g_free(udata);
 }
 
 /*
@@ -1002,7 +1026,7 @@ file_selection_ok_cb(GtkWidget *w,
         gtk_widget_hide(cbs_ptr->fntcol_dialog);
         gtk_entry_set_text(GTK_ENTRY(cbs_ptr->entry), filename);
         gtk_widget_destroy(cbs_ptr->fntcol_dialog);
-        (void) g_free(udata);
+        g_free(udata);
     }
 }
